### PR TITLE
Fix LedgerStatusError and retry on such error

### PR DESCRIPTION
**Pull request by Kelvin-M**
As we use a distributed ledger, state of the ledger could not be the same at a time on each node.
For instance, between two nodes

```
WORKER - NODE 1
INFO 2020-02-28 10:24:46,653 substrapp.ledger_utils 5024 140147702019904 smartcontract invoke:logStartAggregate; elaps=298.21ms; error=LedgerError 

CHAINCODE - NODE 2
[BLOCK 3736]
2020-02-28 10:24:46.181 UTC [substra-chaincode] Infof -> INFO 2388[0m aggregatetuple a4a44bff613ab053825c2185b168fb1c4985cfa764721c9051dcea3718d46eee status updated: todo (from=waiting)
2020-02-28 10:24:46.185 UTC [substra-chaincode] Infof -> INFO 2389[0m Response from chaincode: main.outputCompositeTraintuple{Key:"42312efd79632cf0e9167824a86d3a3a3ffbeb36d62f6f2cbc2b2c2abc9a4c70", Algo:(*main.HashDressName)(0xc0001900f0), Creator:"Org1MSP", Dataset:(*main.outputTtDataset)(0xc0004886c0), ComputePlanID:"8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f", InHeadModel:(*main.Model)(0xc0004caba0), InTrunkModel:(*main.Model)(0xc0004cacf0), Log:"", OutHeadModel:main.outHeadModelComposite{OutModel:(*main.Hash)(0xc0003e0f60), Permissions:main.outputPermissions{Process:main.Permission{Public:false, AuthorizedIDs:[]string{"Org7MSP"}}}}, OutTrunkModel:main.outModelComposite{OutModel:(*main.HashDress)(0xc00127e660), Permissions:main.outputPermissions{Process:main.Permission{Public:false, AuthorizedIDs:[]string{"Org1MSP"}}}}, Rank:45, Status:"done", Tag:""}, error: %!s(<nil>)
[NEW PROPOSAL ]
2020-02-28 10:24:46.474 UTC [substra-chaincode] Infof -> INFO 238a[0m Args received by the chaincode: []string{"logStartAggregate", "{\"key\": \"a4a44bff613ab053825c2185b168fb1c4985cfa764721c9051dcea3718d46eee\"}"}
2020-02-28 10:24:46.627 UTC [substra-chaincode] Infof -> INFO 238b[0m Response from chaincode: main.outputAggregatetuple{Key:"", Algo:(*main.HashDressName)(nil), Creator:"", ComputePlanID:"", Log:"", InModels:[]*main.Model(nil), OutModel:(*main.HashDress)(nil), Rank:0, Status:"", Tag:"", Permissions:main.outputPermissions{Process:main.Permission{Public:false, AuthorizedIDs:[]string(nil)}}, Worker:""}, error: update aggregatetuple a4a44bff613ab053825c2185b168fb1c4985cfa764721c9051dcea3718d46eee failed: cannot change status from waiting to doing

PEER - NODE 2
2020-02-28 10:24:46.255 UTC [committer.txvalidator] Validate -> INFO 8a5b [mychannel] Validated block [3736] in 4ms
2020-02-28 10:24:46.635 UTC [kvledger] CommitWithPvtData -> INFO 8a60 [mychannel] Committed block [3736] with 1 transaction(s) in 379ms (state_validation=3ms block_and_pvtdata_commit=256ms state_commit=114ms) commitHash=[39273e371e9c304ccdd37e948576710726efb998570592e9cc1c7dc8d42677ab]
```

The NODE 1 won't manage to do a `logStartAggregate`, because it can't endorsed its proposal with the NODE2
We can see that, NODE 2 receive block 3736, which has been validated at 10:24:46.255, at  10:24:46.635.
But, the endorsement of the invoke of the proposal  `logStartAggegate` is triggered at 10:24:46.474  from NODE 1 and refused at 10:24:46.627 by NODE 2

There is 161ms state difference between NODE 1 and NODE2 (receive of block 3736, and validation the next block proposal) can be a issue and prevent the compute plan to keep on

In order to fix this issue, we can retry on `LedgerStatusError` to try to avoid those errors caused by state difference (that should not be more than 1s if nodes are connected and try to be sync to the final ledger state)

### CODE REVIEW

File: backend/substrapp/ledger_utils.py [hunk 1]
```diff
@@ -119,7 +124,8 @@ def retry_on_error(delay=1, nbtries=15, backoff=2, exceptions=None):
         RpcError,
         LedgerUnavailable,
         LedgerPhantomReadConflictError,
-        LedgerEndorsementPolicyFailure
+        LedgerEndorsementPolicyFailure,
+        LedgerStatusError,
```
Comment (samlesu) [inline backend/substrapp/ledger_utils.py]: Could you please explain briefly in the code as a comment or in the PR description why we retry on a `LedgerStatusError` ?

I'm quite sure that we won't remember in a couple of weeks / months why this is required :)

File: backend/substrapp/ledger_utils.py [hunk 2]
```diff
@@ -106,8 +106,13 @@ def _raise_for_status(response):
     """Parse ledger response and raise exceptions in case of errors."""
     if not response or 'error' not in response:
         return
+
+    if 'cannot change status' in response['error']:
+        raise LedgerStatusError.from_response_dict(response)
```
Comment (AurelienGasser) [inline backend/substrapp/ledger_utils.py]: @Kelvin-M quick sanity check: did you test that this `raise` is indeed executed?

Comment (Kelvin-M) [reply backend/substrapp/ledger_utils.py]: Yes :)

``` python
response = _call_ledger('invoke', fcn, args)
[response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb5599
2cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\
003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"erro
r\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb5599
2cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\
003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"erro
r\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
]
---------------------------------------------------------------------------
LedgerStatusError                         Traceback (most recent call last)
<ipython-input-9-eaddb07d6156> in <module>
----> 1 response = _call_ledger('invoke', fcn, args)

<ipython-input-8-2f7be6703c84> in _call_ledger(call_type, fcn, args, kwargs)
     76
     77         # Raise errors if status is not ok
---> 78         _raise_for_status(response)
     79
     80         return response

<ipython-input-7-d92d7dbbabbc> in _raise_for_status(response)
      5
      6     if 'cannot change status' in response['error']:
----> 7         raise LedgerStatusError.from_response_dict(response)
      8
      9     status_code = response['status']

LedgerStatusError: update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing
```

Comment (AurelienGasser) [reply backend/substrapp/ledger_utils.py]: :propre:

File: backend/substrapp/ledger_utils.py (hunk 1 above)
Comment (Kelvin-M) [reply backend/substrapp/ledger_utils.py]: Done

**Pull request merged**